ssl/statem/statem_lib.c

```diff
@@ -1227,14 +1226,15 @@ int tls_get_message_body(SSL *s, size_t *len)
         n -= readbytes;
     }
 
-#ifndef OPENSSL_NO_NEXTPROTONEG
     /*
      * If receiving Finished, record MAC of prior handshake messages for
      * Finished verification.
      */
-    if (*s->init_buf->data == SSL3_MT_FINISHED)
-        ssl3_take_mac(s);
-#endif
+    if (*s->init_buf->data == SSL3_MT_FINISHED && !ssl3_take_mac(s)) {
```

Comment:
I found `*s->init_buf->data` a bit hard to read. `*(s->init_buf->data)` or `s->init_buf->data[0]` would be better. This is somewhat of a side note, so "bug off" is acceptable answer, in which case I'd be ready to approve...

Reply:
I went with `*(s->init_buf->data)`. Both here and in #5286.